scrapers/scraper.py

```diff
@@ -0,0 +1,434 @@
+                break
+            state = "unknown"
+
+        data.append(
+            {
+                "lang": language,
+                "state": state,
+                "stems": stems,
+                "location": "{} ({})".format(dirName, location),
+                "contributors": contributors,
+            }
+        )
+
+    return data
+
+
+def pairData(packages):
+    """Returns the locations and stems of all specified pairs"""
+    data = []
+    for package in packages:
+        if not re.match(r"apertium-\w+-\w+", package["name"]):
+            continue
+
+        pairName = package["name"].replace("apertium-", "")
+        pairSet = set(pairName.split("-"))
+        if not pairSet <= set(languages):
+            continue
+        for topic in package["topics"]:
+            if topic[len("apertium-") :] in pairLocations:
+                location = topic[len("apertium-") :]
+                break
+        stems = json.loads(
+            requests.get(
+                "https://apertium.projectjj.com/stats-service/apertium-{}/bidix".format(
+                    pairName
+                )
+            ).text
+        )["stats"][0]["value"]
+        data.append({"langs": list(pairSet), "location": location, "stems": stems})
+
+    return data
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Scrape data for visualizer")
+    parser.add_argument(
+        "-s",
+        "--shallow",
+        help="Faster mode, doesn't dig through histories",
+        action="store_true",
+    )
+    parser.add_argument(
+        "-u",
+        "--updatemailmap",
+        help="outputs users that aren't on .mailmap",
+        action="store_true",
+    )
+    args = parser.parse_args()
+
+    logging.getLogger("countStems").setLevel(logging.CRITICAL)
+
+    dirPath = Path(__file__).absolute().parent.parent
+
+    allPackages = json.loads(
+        requests.get("https://apertium.projectjj.com/stats-service/packages").text
+    )["packages"]
+    pairsFile = open(
+        os.path.join(dirPath, "json", "pairData.json"), "w+", encoding="utf-8"
+    )
+    json.dump(pairData(allPackages), pairsFile, ensure_ascii=False)
+    langsFile = open(
+        os.path.join(dirPath, "json", "transducers.json"), "w+", encoding="utf-8"
+    )
+    json.dump(
+        monoData(allPackages, args.updatemailmap, dirPath),
+        langsFile,
+        ensure_ascii=False,
+    )
+    for lang in languages:
+        if not args.shallow:
+            langHistory = []
+            langHistory.append(monoHistory(lang, dirPath))
+            langHistory.extend(pairHistory(lang, allPackages, dirPath))
+            outputFile = open(
+                os.path.join(dirPath, "json", "{}.json".format(lang)),
+                "w+",
+                encoding="utf-8",
+            )
+            json.dump(langHistory, outputFile, ensure_ascii=False)
+    for dir in next(os.walk(os.path.join(dirPath, "scrapers")))[1]:
+        if "apertium-" in dir:
```

Comment:
Does this delete all the git repos? How does this work with the not-cloning unless present?